### DISCUSSION TITLE
cannot find module Route

### DISCUSSION THREAD

**Discussion by mustard-mh** [nanoexpress/nanoexpress#237] [Q&A] [answered]
## Bug Report

use route feature then  run `node index.js` throw error: Cannot find module '.../nanoexpress/src/Route'
> $projectDict/node_modules/nanoexpress/src/Route.js is exist

(What exactly [PRO PRO Slim version](https://nanoexpress.js.org/routes/route) mean?)

### Is you/your team sponsoring this project

- [ ] Yes
- [x] No

### Minimal reproducible repo

```javascript
// file: index.js
const nanoexpress = require('nanoexpress')
const Route = require('nanoexpress/src/Route')
const app = nanoexpress()

const testRoute = new Route()
app.use('/test', testRoute)
testRoute.get('/', async () => 'hello')

app.listen(3000).then().catch(e => console.error)
```

### Current Behavior

failed to run project

### Expected behavior/code

service start with route

### Environment

- node -v: `v16.2.0`
- OS: macOS Monterey 12.0 Beta

**Comment by dalisoft** [comment, marked as answer]
Hi @mustard-mh 

Thank you for trying out nanoexpress and reporting bugs or any questions related to documentation and it's improves documentation base.
I am updated documentation.

Try this
```diff
- const nanoexpress = require('nanoexpress')
+ const nanoexpress = require('nanoexpress/cjs')
- const Route = require('nanoexpress/src/Route')
+ const Route = require('nanoexpress/cjs/Route')
const app = nanoexpress()

const testRoute = new Route()
app.use('/test', testRoute)
testRoute.get('/', async () => 'hello')

- app.listen(3000).then().catch(e => console.error)
+ app.listen(3000).then().catch(e => console.error(e))
```

I hope this helps you

---

About **PRO and PRO Slim**, before has been three versions. I am removed simple version and made **PRO** version as **FREE** in hope of community supports me by code contribution, financial or documentation/tests improvements.
Seems works, but very slow, but good peoples like you helps project grow by reporting project.

Now only one version - **nanoexpress** itself, before known as **PRO**

**Reply by soulsam480** [reply]
Hello, having the same issue in typescript. 
```
Could not find a declaration file for module 'nanoexpress/src/Route'. '/home/zoro/projects/personal/junev2/node_modules/nanoexpress/src/Route.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/nanoexpress` if it exists or add a new declaration (.d.ts) file containing `declare module 'nanoexpress/src/Route';`ts(7016)
```

**Reply by dalisoft** [reply]
> Hello, having the same issue in typescript.
> 
> ```
> Could not find a declaration file for module 'nanoexpress/src/Route'. '/home/zoro/projects/personal/junev2/node_modules/nanoexpress/src/Route.js' implicitly has an 'any' type.
>   Try `npm i --save-dev @types/nanoexpress` if it exists or add a new declaration (.d.ts) file containing `declare module 'nanoexpress/src/Route';`ts(7016)
> ```

Your case related to Typescript declarations which is currently you can only fix this. 
This case was runtime issue

**Reply by mustard-mh** [reply]
works with
```diff
- const nanoexpress = require('nanoexpress/cjs')
+ const nanoexpress = require('nanoexpress')
- const Route = require('nanoexpress/cjs/Route')
+ const Route = require('nanoexpress/cjs/Route.js')
```

**Reply by mustard-mh** [reply]
@soulsam480  with ts

```typescript
import * as nanoexpress from 'nanoexpress'
import * as Route from 'nanoexpress/cjs/Route.js'
const app = nanoexpress()

const testRoute = new Route()
app.use('/test', testRoute)
testRoute.get('/hello', async () => 'hello')

app.listen(3000).then().catch(console.error)
```

**Reply by mustard-mh** [reply]
@dalisoft  

didn't throw an error with `EADDRINUSE` when the port was used

**Reply by dalisoft** [reply]
@mustard-mh Try reproduce issue in uWebSockets.js itself please, if on uWebSockets.js throws, then let me know and I’ll look into issue

**Reply by ohmree** [reply]
> Hello, having the same issue in typescript.
> 
> ```
> Could not find a declaration file for module 'nanoexpress/src/Route'. '/home/zoro/projects/personal/junev2/node_modules/nanoexpress/src/Route.js' implicitly has an 'any' type.
>   Try `npm i --save-dev @types/nanoexpress` if it exists or add a new declaration (.d.ts) file containing `declare module 'nanoexpress/src/Route';`ts(7016)
> ```

As far as I can tell that's because there are no type declarations for the `Route` class, in order to fix it someone will have to write a corresponding `.d.ts` file or rewrite `Route.js` in typescript.

This error (or warning, depending on your config) is harmless and you can safely ignore it but you'll have worse intellisense and no typechecks for any code that interfaces with these parts of nanoexpress.

**Comment by KH93b** [comment]
const Route = require('../node_modules/nanoexpress/cjs/Route.cjs');
const route = new Route();
app.use(route);
route.get('/', async () => 'hello world');

i am getting error TypeError: Cannot read properties of undefined (reading 'get')
the error occurs if you export the route from the module.

**Reply by dalisoft** [reply]
@KH93b Please try `import Route from 'nanoexpress/Route'`
